### PR TITLE
#3704 Avoid sync leap just before upgrade

### PR DESCRIPTION
**Pull request by EdHastingsCasperAssociation**
as the title says, and also fixed an incorrect state change to validate mode for non-validators present at genesis which was revealed by tighter leap instruction tracking and testing

### CODE REVIEW

**Comment by rafal-ch** [comment]
bors try

**Review by Fraser999 — APPROVED** [review]

File: node/src/components/block_accumulator/leap_instruction.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    pub(super) fn from_execution_threshold(
+        attempt_execution_threshold: u64,
+        distance_from_highest_known_block: u64,
+        is_upgrade_boundary: bool,
+    ) -> Self {
+        if distance_from_highest_known_block == 0 {
+            return LeapInstruction::AtHighestKnownBlock;
+        }
+        if is_upgrade_boundary
+            && distance_from_highest_known_block <= attempt_execution_threshold * 2
```
Comment (rafal-ch) [inline node/src/components/block_accumulator/leap_instruction.rs:28]: May be worthwhile to explain in a comment why we think `* 2` is enough.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: 
No blockers.

**Pull request closed without merging**